sky/packages/sky/lib/widgets/switch.dart

```diff
@@ -20,52 +25,124 @@ const double _kThumbRadius = 10.0;
 const double _kTrackHeight = 14.0;
 const double _kTrackRadius = _kTrackHeight / 2.0;
-const double _kTrackWidth = _kSwitchWidth - (_kThumbRadius - _kTrackRadius) * 2.0;
+const double _kTrackWidth =
+    _kSwitchWidth - (_kThumbRadius - _kTrackRadius) * 2.0;
+const Duration _kCheckDuration = const Duration(milliseconds: 200);
+
+class Switch extends Component {
+  Switch({Key key, this.value, this.onChanged}) : super(key: key);
+
+  bool value;
+  ValueChanged onChanged;
+
+  Widget build() {
+    return new SwitchWrapper(
+        value: value,
+        onChanged: onChanged,
+        thumbColor: Theme.of(this).accentColor);
+  }
+}
 
-class Switch extends Toggleable {
-  // TODO(jackson): Hit-test the switch so that it can respond to both taps and swipe gestures
+// This wrapper class exists only because Switch needs to be a Component in
+// order to get an accent color from a Theme but Components do not know how to
+// host RenderObjects.
+class SwitchWrapper extends OneChildRenderObjectWrapper {
+  SwitchWrapper({Key key, this.value, this.onChanged, this.thumbColor})
+      : super(key: key);
 
-  Switch({
-    Key key,
-    bool value,
-    ValueChanged onChanged
-  }) : super(key: key, value: value, onChanged: onChanged);
+  bool value;
+  ValueChanged onChanged;
+  Color thumbColor;
+
+  RenderSwitch get root => super.root;
+  RenderSwitch createNode() => new RenderSwitch(value, onChanged, thumbColor);
+
+  void syncRenderObject(SwitchWrapper old) {
+    super.syncRenderObject(old);
+    root.value = value;
+    root.thumbColor = thumbColor;
+    root.onChanged = onChanged;
+  }
+}
+
+class RenderSwitch extends RenderProxyBox {
+  RenderSwitch(this._value, this.onChanged, this._thumbColor) {
+    _position = new AnimatedValue<double>(0.0, end: 1.0);
+    _performance = new AnimationPerformance()
+      ..variable = position
+      ..duration = _kCheckDuration
+      ..progress = _value ? 1.0 : 0.0
+      ..addListener(markNeedsPaint);
+  }
+
+  void handleEvent(sky.Event event, BoxHitTestEntry entry) {
+    if (event is sky.GestureEvent && event.type == 'gesturetap')
+      onChanged(!_value);
+  }
 
   Size get size => const Size(_kSwitchWidth + 2.0, _kSwitchHeight + 2.0);
+  bool _value;
+  ValueChanged onChanged;
+  Color _thumbColor = _kThumbOffColor;
+
+  void set value(bool value) {
+    if (value == _value) return;
+    _value = value;
+    // TODO(abarth): Setting the curve on the position means there's a
+    // discontinuity when we reverse the timeline.
+    if (value) {
+      position.curve = easeIn;
+      performance.play();
+    } else {
+      position.curve = easeOut;
+      performance.reverse();
+    }
+  }
 
-  void customPaintCallback(sky.Canvas canvas, Size size) {
+  void set thumbColor(Color value) {
+    if (value == _thumbColor) return;
+    _thumbColor = value;
+    markNeedsPaint();
+  }
+
+  AnimatedValue<double> _position;
+  AnimatedValue<double> get position => _position;
```

Comment:
This pattern can be replaced by `final`

```
final AnimatedValue<double> _position = new AnimatedValue<double>(0.0, end: 1.0);
```

Then you can drop the code from the constructor.